presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -901,6 +913,17 @@ public boolean supportsLimit()
         return Optional.empty();
     }
 
+    @Override
+    public boolean supportsTopNPushDown()
+    {
+        return orderByFunction().isPresent();
+    }
+
+    protected Optional<BiFunction<String, List<SortItem>, String>> orderByFunction()
+    {
+        return Optional.empty();
```

Comment:
Would it make sense to have a default implementation?

Reply:
Not really, I looked at mysql (doesen' support nulls first/last) same with sql server.

if anything we should probably make limit function default which is pretty much same across all jdbc clients other than pheniox.